src/Settings/SettingsGuiHelper.cpp

```diff
@@ -261,6 +261,7 @@ void SettingsGuiHelper::updateParameters(const QJsonObject &data)
         val = data["value"].toBool();
     }
     m_settings->updateParam(m_settings->getParamId(param), val);
+    m_mw->checkSettingsChanged();
```

Comment:
out of curiosity, what's that for?

Reply:
src/utils/KeyboardLayoutDetector.cpp:23 I am sending parameters directly for layouts, it updates these parameters in daemon and send back param_changed message, which calls this function and setting UI element to correct value (in this case the layout). However settingsChanged is not checked and due to the new layout Save/Cancel buttons are displayed without this call.